services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker.py

```diff
@@ -65,6 +57,7 @@ def on_app_shutdown(app: FastAPI) -> Callable[[], Awaitable[None]]:
     async def _stop() -> None:
         if app.state.resource_tracker_rabbitmq_consumer:
             await _unsubscribe_from_rabbitmq(app)
+            await app.state.rabbitmq_client.close()
```

Comment:
ok I see now what you are doing here. but is there not something wrong in the order which you define the startup/shutdown? I think fastapi shall keep the ordering @pcrespov?

Reply:
From the try/error approach I am not sure whether the ordering is guaranteed. This is my order in application.py
```
    setup_rabbitmq(app)

    setup_resource_tracker(app)
```

Reply:
From the documentation in FastAPI they dot say. But I think that since we add the event, they are probably run in sequence, meaning that the shutdown events are run one after the others. Which mean the order is broken, which would explain your problem.
So Maybe please add an issue and a link to fix it once FastAPI allows to have several lifespan events. Otherwise, we need to implement our own.

Reply:
I will create an issue :+1: